datalad/core/distributed/push.py

```diff
@@ -266,6 +265,23 @@ def __call__(
             for i, ds in pbars.items():
                 log_progress(lgr.info, i, 'Finished push of %s', ds)
         if not matched_anything:
+            potential_remote = False
+            if not to and path:
+                # if we get a remote name without --to, provide a hint
+                sr = ds_repo.get_remotes(**get_remote_kwargs)
+                potential_remote = [
+                    p for p in ensure_list(path) if p in sr
+                ]
+            if potential_remote:
+                msg = "It seems like you specified a sibling name {} " \
+                      "as a path. Forgot to use --to?".format(potential_remote)
+                yield dict(
+                    res_kwargs,
+                    status='impossible',
+                    message=msg,
+                    type='dataset',
+                    path=ds.path,
+                )
```

Comment:
Unclear to me, if the next `yield` should not move into an `else:`. If not, a comment on why not might help the next reader.

Reply:
I appreciate hints on what is considered best practice in this case. The next yield, I think, should only be conditional on ``not matched_anything``. The previous conditionals (`ìf not to and path`` and ``potential_remote``) only exist to capture the UX issue this PR tries to tackle before the "notneeded" of the next yield statement is issued.
I could do an ``else [no potential_remote]``, but to me this feels like I would unnecessarily confine myself into a conditional (i.e., somewhat implying that ``potential_remote`` is more related to unmatched paths)... What would you do? Is it more readable to use ``else``?

Reply:
Looking at this again, I'd say there are two valid things:
- `else`
- `return` after the `impossible` result

Currently both results are yielded - which we didn't catch before, since it's not tested and a command line output wouldn't show it b/c `notneeded` isn't rendered by default.

Re style: I'd probably go with `return` and a comment before the next yield that states the condition we are in at that point (not matched anything). That's b/c of the same feeling: The `notneeded` has a relation to the conditional for `impossible` only "by chance" at this point in time (could change). But that's no strong opinion.

Reply:
thx, done